src/CMakeLists.txt

```diff
@@ -33,6 +33,14 @@ endif ()
 if (REST)
     pkg_check_modules(SOUP REQUIRED libsoup-2.4)
     pkg_check_modules(QRENCODE REQUIRED libqrencode)
+    set(REST_PACKAGES
+        libsoup-2.4
+    )
+
+    if (${CMAKE_HOST_WIN32})
+        # Needed for Posix.gethostname()
```

Comment:
Is [Resolver.lookup_by_name](https://valadoc.org/gio-2.0/GLib.Resolver.lookup_by_name.html) more OS-independant?

Reply:
Hmm, maybe I miss something, but `gethostname` is to get _our own_ hostname (and then, IP). It's still a very unreliable way, but there is nothing better in Vala, I believe. Maybe we can add some code in C (which will not be OS-independent, either).

Reply:
Oh, and there is the whole [NetworkManager API](https://valadoc.org/libnm/index.htm), but OS neutrality is, of course, very questionable.

Reply:
Maybe Posix is then the smaller evil compared to NetworkManager :)